### ISSUE TITLE
Export Meter identified as import plus Import Data stopped updating after updating to 13.0.2

### ISSUE THREAD

**Issue by jwt99412** [BottlecapDave/HomeAssistant-OctopusEnergy#1041]
### Describe the bug

This is almost certainly to do with Octopus but the integration identifies my export meter as an import meter after meter change.
I have raised it at the Octopus end but wanted to check here as well.
The main problem is that, since updating the integration to 13.0.2 the import meter data has stopped updating. Latest data was for 17/10 just before I updated the integration.

See attached diagnostic setup file.
[octopus_energy-01JAP11SA4VE26TRQ8HSCXH9ME-Electricity Meter-b82db768bb1852fb1f363978d51062d4.json](https://github.com/user-attachments/files/17458560/octopus_energy-01JAP11SA4VE26TRQ8HSCXH9ME-Electricity.Meter-b82db768bb1852fb1f363978d51062d4.json)

### Reproduction steps

Set up the integration from scratch with my account details (available on request)
Three meters discovered, two electricity meters marked as import plus gas meter
Rates are updated for all meters showing that the export meter tariff is correct but marked as import.
I waited 3 days but since updating the integration the electricity import data has not been updated.

I also deleted and reinstalled the integration in case the old meter data was corrupting things and this made no difference.

### Expected behaviour

(After Octopus have fixed their end - not hopeful) the export meter should be correctly identified.
Expect import and export data to be updated in line with the Octopus data and separately the Octopuswatch app data which is collecting the import data after the HA integration has stopped.

### Tariff Code

E-1R-FLUX-EXPORT-23-02-14-A

### Integration Version

13.0.2

### Home Assistant Version

Core 2024.9.3 Supervisor 2024.10.2 Operating System 13.1 Frontend 20240909.1

### Fresh Install?

After upgrading

### Home Assistant Logs

Can't see any log entries

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by jwt99412** [comment]
Octopus have now updated the meter type flag to export for the export meter. Import data available on website up to 20/10. Last available data timestamp is 19/10 at 01:30. Last 30min data available up to 18:10 just after midnight.

**Comment by jwt99412** [comment]
Some detailed logs
[home-assistant_octopus_energy_2024-10-23T08-59-50.188Z.log](https://github.com/user-attachments/files/17488095/home-assistant_octopus_energy_2024-10-23T08-59-50.188Z.log)

**Comment by jwt99412** [comment]
Reading the documentation for the new version it says the date of validity attribute was removed in favour of separate validity sensors. I note that the import sensors all still have the time of validity attribute on them. Could this be preventing them being updated? The Export sensors have started updating and do not have the time of validity attribute on there.
The previous accumulative consumption sensors cannot be deleted as far as I can see so no way to reset them.

**Comment by jwt99412** [comment]
Tried to refresh import statistics via the refresh service got error as follows:

2024-10-23 21:04:49.044 ERROR (MainThread) [homeassistant.components.automation.refresh_import_consumption] While executing automation automation.refresh_import_consumption
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 755, in async_trigger
    return await self.action_script.async_run(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1795, in async_run
    return await asyncio.shield(create_eager_task(run.async_run()))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 463, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 527, in _async_step
    self._handle_exception(
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 557, in _handle_exception
    raise exception
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 525, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 763, in _async_call_service_step
    response_data = await self._async_run_long_action(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 726, in _async_run_long_action
    return await long_task
           ^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2761, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2804, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 996, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1068, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/octopus_energy/electricity/previous_accumulative_consumption.py", line 197, in async_refresh_previous_consumption_data
    await async_refresh_previous_electricity_consumption_data(
  File "/config/custom_components/octopus_energy/statistics/refresh.py", line 76, in async_refresh_previous_electricity_consumption_data
    previous_consumption_result = await async_import_external_statistics_from_consumption(
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/octopus_energy/statistics/consumption.py", line 102, in async_import_external_statistics_from_consumption
    peak_totals[peak_type] = peak_statistics[-1]["sum"] if peak_statistics[-1] is not None else 0
                                                           ~~~~~~~~~~~~~~~^^^^
IndexError: list index out of range

**Comment by BottlecapDave** [comment]
Wow, that was a pain to debug. I believe I've fixed your issue in https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/v13.0.3.

Thank you for all of the logs that you supplied. It turns out your import meter was reporting more data than most other meters. During BST, the consumption endpoint always typically returns the first hour of the current day. The previous consumption sensor takes this into account by requesting 50 records instead of 48 records (number of 30 minute periods in a day). However, for some reason your import meter is reporting 1.5 hours of the current day. 

In v13.0.3 I've adjusted the number of records being requested to take this into account, which I'm hoping will fix your issue. I've also included some additional diagnostic and logging information in case it doesn't to help diagnose your issue.
